agents/ntp/ntp.py

```diff
@@ -36,27 +36,48 @@ def service_healthy(
             )
 
     def maybe_restart_service(self):
+        """
+        Check NTP and SSH services; if any are inactive, attempt restart.
+        Returns True if all services are healthy (or successfully restarted),
+        False if one failed to restart.
+        """
         inactive_services = []
 
+        # SSH
         if not self.is_ssh_service_active():
             inactive_services.append('ssh')
 
-        if inactive_services:
-            for service in inactive_services:
-                restart_service(self.logger, self.fallback_logger, service)
+        # NTP daemons
+        # If none of the configured critical_processes are running,
+        # treat as down.
+        ntp_running = False
+        for proc in self.critical_processes:
+            self.processes = [proc]
```

Comment:
This alters `self.processes`, which might affect future checks or logic. To prevent unintended side effects, consider modifying `_is_process_running()` to accept a process name directly, e.g.:
```
def _is_process_running(self, proc_name=None):
    ...
```
And then use:
```
if self._is_process_running(proc):
```